docroot/modules/humsci/hs_capx/src/Capx.php

```diff
@@ -0,0 +1,262 @@
+<?php
+
+namespace Drupal\hs_capx;
+
+use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\Database\Connection;
+use Drupal\Core\Logger\LoggerChannelFactory;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\Exception\GuzzleException;
+
+/**
+ * Service to test CAPx connection and import organization data.
+ */
+class Capx {
+
+  /**
+   * API url used for organization data.
+   */
+  const API_URL = 'https://api.stanford.edu';
+
+  /**
+   * Authentication url.
+   */
+  const AUTH_URL = 'https://authz.stanford.edu/oauth/token';
+
+  /**
+   * The actual CAP API.
+   */
+  const CAP_URL = 'https://cap.stanford.edu/cap-api/api/profiles/v1';
```

Comment:
Do we need to plan for the possibility of a v2 endpoint or would we just hardwire the new API endpoint version to the module when we are ready? Do we expect older versions of this module to work with a new API schema? Probably not, so this is likely ok.

Reply:
i'd guess that a V2 would require refactoring of the api calls so i think this should be safe.